### PR TITLE
Add extrusion sensors for patched CC

### PR DESCRIPTION
**Pull request by harpua555**
Introduces 'total_extrusion' and 'current_extrusion' sensor entity descriptions in definitions.py. Updates PrintInfo model to parse and expose total and current extrusion values, handling both standard and hex-encoded keys from printer data.  This will only be needed for the new patch PR in OCC FW (exposes these two new JSON key/value pairs).  If these pairs don't exist, the sensors are inactive.

## Summary by CodeRabbit

* **New Features**
  * Added two extrusion-tracking sensors:
    * Total extrusion — cumulative distance (mm).
    * Current extrusion — instantaneous extrusion (mm).
  * Both sensors report values with units and precision for improved print monitoring.

* **Tests**
  * Added tests covering extraction and edge cases (different firmware encodings, zero vs missing values).

### CODE REVIEW

File: custom_components/elegoo_printer/sdcp/models/status.py [hunk 1]
```diff
@@ -160,6 +162,20 @@ def __init__(
...
+        self.total_extrusion: float | None = (
+            data.get("TotalExtrusion")
+            if "TotalExtrusion" in data
+            else data.get("54 6F 74 61 6C 45 78 74 72 75 73 69 6F 6E 00")  # Hex for "TotalExtrusion"
+        )
+        self.current_extrusion: float | None = (
+            data.get("CurrentExtrusion")
+            if "CurrentExtrusion" in data
+            else data.get("43 75 72 72 65 6E 74 45 78 74 72 75 73 69 6F 6E 00")  # Hex for "CurrentExtrusion"
+        )
```
[the remark this answers is not in the document]
Comment (danielcherubini) [reply custom_components/elegoo_printer/sdcp/models/status.py:180]: @harpua555 can you update the tests?

File: custom_components/elegoo_printer/sdcp/models/status.py [hunk 2]
```diff
@@ -160,6 +162,20 @@ def __init__(
...
+        # Use explicit None check to handle 0 values correctly
+        self.total_extrusion: float | None = (
+            data.get("TotalExtrusion")
+            if "TotalExtrusion" in data
+            else data.get("54 6F 74 61 6C 45 78 74 72 75 73 69 6F 6E 00")  # Hex for "TotalExtrusion"
+        )
+        self.current_extrusion: float | None = (
+            data.get("CurrentExtrusion")
+            if "CurrentExtrusion" in data
+            else data.get("43 75 72 72 65 6E 74 45 78 74 72 75 73 69 6F 6E 00")  # Hex for "CurrentExtrusion"
```
Comment (danielcherubini) [inline custom_components/elegoo_printer/sdcp/models/status.py]: line too long, please run `make lint`

File: custom_components/elegoo_printer/sdcp/models/status.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (harpua555) [reply custom_components/elegoo_printer/sdcp/models/status.py:180]: > @harpua555 can you update the tests?

The potential issue is correct - the JSON does actually output raw space separated bytes (for now)

As far as the failed ruff, it looks like it is complaining about line length.  Is this a self-imposed limit or something HA imposes?  I am having no issues w/ the sensors on my HA instance as PRed

[the remark this answers is not in the document]
Comment (harpua555) [reply custom_components/elegoo_printer/sdcp/models/status.py:180]: Just realized that was the linter - updated the .py - linter should be happy now

[the remark this answers is not in the document]
Comment (harpua555) [reply custom_components/elegoo_printer/sdcp/models/status.py:180]: rabbit should also be happy now - rerunning

**Review by danielcherubini — APPROVED** [review]

**Comment by danielcherubini** [comment]
Thanks heaps!

**Comment by danielcherubini** [comment]
one last question.. did you test these yourself?..

**Comment by harpua555** [comment]
Yep - I spun up a dev HA instance w/ the makefile and did some prints.  Total extrusion and current extrusion both show expected values, and total autoclears once a new print is started.  Have a WIP spool manager that is using these as well.

image

image

image

**Comment by danielcherubini** [comment]
Very very cool. 

Ok I'll get a release out with this in it soon.
